dss/operations/elasticsearch.py

```diff
@@ -31,13 +32,34 @@
                                  "--keys": dict(default=None, nargs="*", help="keys to index.")})
 def index(argv: typing.List[str], args: argparse.Namespace):
     """
-    Initiate the indexer lambda for `keys`
+    Queue an SQS message to the indexer lambda for each key in `keys`.
     """
     index_queue_url = get_queue_url("dss-index-operation-" + os.environ['DSS_DEPLOYMENT_STAGE'])
     with SQSMessenger(index_queue_url) as sqsm:
         for key in args.keys:
             sqsm.send(json.dumps(dict(replica=args.replica, key=key)))
 
+@elasticsearch.action("index-prefix",
+                      arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
+                                 "--prefix": dict(default=None, help="UUID prefix to index.")})
+def index_prefix(argv: typing.List[str], args: argparse.Namespace):
+    """
+    Queue an SQS message to the indexer lambda for each key in `bundles/{prefix}`.
+    """
+    replica = Replica[args.replica]
+    handle = Config.get_blobstore_handle(replica)
+    index_queue_url = get_queue_url("dss-index-operation-" + os.environ['DSS_DEPLOYMENT_STAGE'])
+
+    def _forward_keys(pfx):
+        with SQSMessenger(index_queue_url) as sqsm:
+            for key in handle.list(replica.bucket, pfx):
+                sqsm.send(json.dumps(dict(replica=args.replica, key=key)))
+
+    with ThreadPoolExecutor(max_workers=10) as e:
+        futures = [e.submit(_forward_keys, f"bundles/{args.prefix}{c}") for c in hexdigits.lower()]
```

Comment:
Would using f"bundles/{args.prefix}{c}" mean that an operator would have to submit multiple calls to this script in order to cover all 22 possible starting values for a `uuid`?

Reply:
Since UUIDs are hexadecimal digits the operator would need to issue 16 calls to cover the entire UUID space.

I'll extend the script to send a message for every bundle if `--prefix` is omitted.